core/chain_makers.go

```diff
@@ -246,6 +246,7 @@ func GenerateChain(config *params.ChainConfig, parent *types.Block, engine conse
 		config = params.TestChainConfig
 	}
 	headers, blocks, receipts := make([]*types.Header, n), make(types.Blocks, n), make([]types.Receipts, n)
+	types.SetHeaderSealFlag(config.IsHeaderWithSeal())
```

Comment:
"global variable" will affect other tests which running in parallel?
`go test -p 8 --count=40 -race --trimpath -run='TestHeader'  ./core` doesn't complain?

Reply:
it basically fine - just wonder if we can do something with it

Reply:
is there any way to access Genesis or ChainConfig from `core/types` package? i did not find anything, so i've realized it with global variable. 
`go test -p 8 --count=40 -race --trimpath -run='TestHeader' ./core` runs ok for me